### ISSUE TITLE
Call givethio endpoint every time a verifid campaign changes in giveth trace

### ISSUE THREAD

**Issue by mohammadranjbarz** [Giveth/giveth-next#364]

**Comment by mohammadranjbarz** [comment]
@divine-comedian 
It's ready for test now.
There is thing that should work 
- [ ] Update title, description in giveth.io should change them in giveth trace
- [ ] Update title, description in giveth trace should change them in giveth.io
- [ ] Archive project in giveth trace would cancel project gievthio 
- [ ] Cancelling project in giveth.io should archive the campaign in giveth trace
- [ ] Activate project in gievth.io should  unArchive project in giveth trace
- [ ] If project Is deactivated in givethio, unArchiving campaign in trace should not change the project to active

**Comment by divine-comedian** [comment]
it wasn`t working with some older projects so I tried to make a new project on giveth.io and make it traceable via the wizard.. however when I try it says the slug is wrong ... can`t screenshot all the details but it looks like this 

![image](https://user-images.githubusercontent.com/67759413/141198472-35cf2e5e-0b0a-4894-81c7-96d78f5697ae.png)

**Comment by mohammadranjbarz** [comment]
> it wasn't working with some older projects so I tried to make a new project on giveth.io and make it traceable via the wizard.. however when I try it says the slug is wrong ... can`t screenshot all the details but it looks like this
> 
> ![image](https://user-images.githubusercontent.com/67759413/141198472-35cf2e5e-0b0a-4894-81c7-96d78f5697ae.png)

Oops you are right, 
changing queries in impact-graph these days, destroyed API calls in giveth trace, I have verified a project just now
https://develop.giveth.io/campaign/test-upload-photo-2

**Comment by divine-comedian** [comment]
checking now it looks like it works, and same with changes on io showing up on trace, it just took some time to load. I will continue testing..

**Comment by divine-comedian** [comment]
ah, and now staging is broken - 

![image](https://user-images.githubusercontent.com/67759413/141209011-da7bc822-36fd-4e1e-a9b5-7b564a1ac4d2.png)

**Comment by divine-comedian** [comment]
okay this is getting weird - I was testing the part of cancelling the project (via adminBro) I look up the project and I have 4 separate entries for the same project - one for probably when the trace was created and then another for each title change 

![image](https://user-images.githubusercontent.com/67759413/141328106-79679564-1172-4c8e-b180-12b43808f629.png)


is this related or shall I make a new issue?

**Comment by mohammadranjbarz** [comment]
@divine-comedian 
Yeah as I talked with @MoeNick  `Archived` is equavalant  to `Cancelled` in giveth.io
( and the business are different we just map them for now)

**Comment by mohammadranjbarz** [comment]
@MoeNick  @DonnaOftadeh 
The problem of these task is resolved, because this week Mitch is in vocation, I would appreciate if someone else take testing of this

**Comment by DonnaOftadeh** [comment]
@mohammadranjbarz @MoeNick  I am taking care of it

**Comment by DonnaOftadeh** [comment]
The first two work fine 
For the Archive: Archive project in giveth trace would cancel project in Giveth.io but it delists it, it should cancel it as you said @mohammadranjbarz 
![image](https://user-images.githubusercontent.com/53400757/141792859-2dbdbb79-7c74-4527-99dc-563fac352ac0.png)

this is the message in io which says it is deactivated (@MoeNick these terminologies are confusing we should figure out something less confusing adn if by cancel we mean deactivated we should use one word across all of our platforms)
![image](https://user-images.githubusercontent.com/53400757/141793132-b77522db-6b97-4f72-adb1-74a46aba02ec.png)

**Comment by DonnaOftadeh** [comment]
@mohammadranjbarz activate the project on io via the admin panel but on trace it is still archived:
Admin panel activated status:
![image](https://user-images.githubusercontent.com/53400757/141795945-58e2598f-fd2e-42b2-aff1-899036404075.png)
io activated project:
![image](https://user-images.githubusercontent.com/53400757/141796131-729ce4f4-06e4-4ab2-ba04-158b0fda5daa.png)
Still archive in Trace:
![image](https://user-images.githubusercontent.com/53400757/141796304-dac67027-49fe-49c7-af0d-c2b2b21aabf3.png)
![image](https://user-images.githubusercontent.com/53400757/141796365-44fa228b-e9a7-4f1f-bcb8-557d5b422512.png)

**Comment by mohammadranjbarz** [comment]
> The first two work fine For the Archive: Archive project in giveth trace would cancel project in Giveth.io but it delists it, it should cancel it as you said @mohammadranjbarz ![image](https://user-images.githubusercontent.com/53400757/141792859-2dbdbb79-7c74-4527-99dc-563fac352ac0.png)
> 
> this is the message in io which says it is deactivated (@MoeNick these terminologies are confusing we should figure out something less confusing adn if by cancel we mean deactivated we should use one word across all of our platforms) ![image](https://user-images.githubusercontent.com/53400757/141793132-b77522db-6b97-4f72-adb1-74a46aba02ec.png)

It's about admin bro that shows the `Cancelled` status as `delisted`, for testing that you can cancel your project with admin bro and you can see that it shows `delisted`]
I think we consider that as a separated issue about admin bro and statuses

**Comment by mohammadranjbarz** [comment]
> @mohammadranjbarz activate the project on io via the admin panel but on trace it is still archived: Admin panel activated status: ![image](https://user-images.githubusercontent.com/53400757/141795945-58e2598f-fd2e-42b2-aff1-899036404075.png) io activated project: ![image](https://user-images.githubusercontent.com/53400757/141796131-729ce4f4-06e4-4ab2-ba04-158b0fda5daa.png) Still archive in Trace: ![image](https://user-images.githubusercontent.com/53400757/141796304-dac67027-49fe-49c7-af0d-c2b2b21aabf3.png) ![image](https://user-images.githubusercontent.com/53400757/141796365-44fa228b-e9a7-4f1f-bcb8-557d5b422512.png)

We should test it together, because I think it works

**Comment by mohammadranjbarz** [comment]
@DonnaOftadeh 
You are right, nice catch.
 We dont handle edits in view project, I just update trace when we use bulk edit in admin bro

**Comment by mohammadranjbarz** [comment]
@DonnaOftadeh 
We deployed a fix for the problem that you said, ( editing project from project view page)
you can test it again

**Comment by DonnaOftadeh** [comment]
@mohammadranjbarz that problem is solved 👍 
but I am testing archive in trace which should deactivate in Giveth - it deactivates but the status id both in the admin panel and in graphql is delisted it should be cancelled
![image](https://user-images.githubusercontent.com/53400757/142416792-eb012d65-be7c-499d-afcb-70edd3a3a47e.png)

**Comment by DonnaOftadeh** [comment]
For the fifth one: Activate project in gievth.io should unArchive project in giveth trace
It doesn't work, it gives this error when I try to activate a traceable project from IO (I unarchived the project from trace platform as activating the project didn't unarchive the project in trace automatically ),it seems that I don't have permission to update the project status:
![image](https://user-images.githubusercontent.com/53400757/142422243-6d9ab6e3-f4c1-41e0-a723-788a3d7a8ee0.png)

**Comment by mohammadranjbarz** [comment]
Thanks for testing
* The statuses description can change in db, so the `cancelled` status name has changed to `delisted` so you can check the s`tatusId`, `7` means cancelled so it's ok
* Deactivate project doesn't change the status of campaign in trace ( just cancelling project archive campaign in trace)
* When you cancel project you can not activate it from user panel, you should activate itjust by admin bro ( users cant activate their cancelled projects)
*

**Comment by DonnaOftadeh** [comment]
@MoeNick can you please clarify the archive, unarchive in trace and cancel, deactivate and delisted in IO, and who can change it, I need a precise description to be able to test the rest of this PR

**Comment by DonnaOftadeh** [comment]
@MoeNick for this one : Cancelling project in giveth.io should archive the campaign in giveth trace
it works when the project status is delisted, but doesn't work when the status is cancelled, please clarify if this is DOD

**Comment by DonnaOftadeh** [comment]
this is tested and it works as wanted, thanks to Mohammad, @mohammadranjbarz sorry for bugging you a lot!
 only the cancelling seems vague: the cancelling task in the bullet list work when the project status is delisted, we have a cancel status which I don't know exactly what is the difference, if they are the same the same action should be applied to cancel status as well
